### PR TITLE
[#60][#42] [iOS] [Integrate] As a user, I can answer my survey

### PR DESCRIPTION
**Pull request by phongvhd93**
- Close:
- #60 
- #42 

## What happened 👀

- Integrate questions and answers from API to `SurveyQuestionView`
- Check if a user is at the last page to show the `Submit` button.
- Add UT and UI test

## Insight 📝

Show the questions and answers from API to the user to allow them to answer their surveys.
## Proof Of Work 📹

https://github.com/nimblehq/avishek-phong-kmm-ic/assets/22606906/145daa1b-fda3-4288-ac69-832e039971f2

### CODE REVIEW

**Comment by markgravity** [comment]
@phongvhd93 please rebase 🙏

File: shared/src/commonMain/kotlin/co/nimblehq/avishek/phong/kmmic/presentation/module/SurveyQuestionViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    private val _viewState: MutableStateFlow<SurveyQuestionViewState> =
+        MutableStateFlow(SurveyQuestionViewState())
+    val viewSate: StateFlow<SurveyQuestionViewState> = _viewState
+
+    private var survey: Survey? = null
+
+    fun updateStateWith(survey: Survey) {
+        this.survey = survey
+        val questionUiModels = survey.questions?.mapIndexed { index, question ->
+                QuestionUiModel(question, "${index + 1}/${survey.questions.count()}")
```
Comment (AVI5HEK) [inline shared/src/commonMain/kotlin/co/nimblehq/avishek/phong/kmmic/presentation/module/SurveyQuestionViewModel.kt]: We should use extension functions as factory methods, e.g., `question.ToQuestionUiModel(index, survey.questions.count())`.

**Review by AVI5HEK — COMMENTED** [review]
We need to merge this PR ASAP. Then I'll remove some of the duplicate UI models I've created in my PRs. 😅

File: shared/src/commonMain/kotlin/co/nimblehq/avishek/phong/kmmic/presentation/uimodel/QuestionUiModel.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import co.nimblehq.avishek.phong.kmmic.domain.model.QuestionDisplayType
+
+data class QuestionUiModel(
+    val id: String,
+    val step: String,
+    val displayOrder: Int,
+    val displayType: QuestionDisplayType,
+    val questionTitle: String,
+    val answers: List<SurveyAnswerUiModel>,
+    var userInputs: Set<AnswerInput>
```
Comment (AVI5HEK) [inline shared/src/commonMain/kotlin/co/nimblehq/avishek/phong/kmmic/presentation/uimodel/QuestionUiModel.kt:17]: Just out of curiosity, why don't we use `List<AnswerInput>` here?

Comment (phongvhd93) [reply shared/src/commonMain/kotlin/co/nimblehq/avishek/phong/kmmic/presentation/uimodel/QuestionUiModel.kt:17]: Using Set will be easier to update the view, especially in the case NPS and radio button answers

File: shared/src/commonMain/kotlin/co/nimblehq/avishek/phong/kmmic/presentation/module/SurveyQuestionViewModel.kt (hunk 1 above)
Comment (phongvhd93) [reply shared/src/commonMain/kotlin/co/nimblehq/avishek/phong/kmmic/presentation/module/SurveyQuestionViewModel.kt]: solved in f97867d

**Review by AVI5HEK — APPROVED** [review]

**Comment by AVI5HEK** [comment]
@phongvhd93 we need to keep the answers in a property in the ViewModel, right? I'm not using a separate ViewModel for the question screens, I'm using the `SurveyDetailViewModel`. For me it'd be easy to keep the provided answers in `SurveyDetailViewModel`. 🤔

**Comment by phongvhd93** [comment]
> @phongvhd93 we need to keep the answers in a property in the ViewModel, right? I'm not using a separate ViewModel for the question screens, I'm using the `SurveyDetailViewModel`. For me it'd be easy to keep the provided answers in `SurveyDetailViewModel`. 🤔

I created a separate ViewModel for the question screen in this [PR](https://github.com/nimblehq/avishek-phong-kmm-ic/pull/136). You need to pass the survey object you got from the detail screen to the question screen. It easier to control the view's state in a separate view model and it won't pollute the view model of the detail screen.

**Comment by AVI5HEK** [comment]
> > @phongvhd93 we need to keep the answers in a property in the ViewModel, right? I'm not using a separate ViewModel for the question screens, I'm using the `SurveyDetailViewModel`. For me it'd be easy to keep the provided answers in `SurveyDetailViewModel`. 🤔
> 
> I created a separate ViewModel for the question screen in this [PR](https://github.com/nimblehq/avishek-phong-kmm-ic/pull/136). You need to pass the survey object you got from the detail screen to the question screen. It easier to control the view's state in a separate view model and it won't pollute the view model of the detail screen.

On the Android app, the same screen is used to fetch the survey details and then populate the questions with the data. 😓

**Review by markgravity — APPROVED** [review]

**Pull request merged**